### PR TITLE
Added hashmap.md in kotlin folder

### LINKED ISSUE #1477: [Concept Entry] Hashmap: Adding hashmap.md in kotlin
## I would like a new concept entry for [Kotlin](https://www.codecademy.com/resources/docs/kotlin) about hashmap

The entry will have:

- Have a section on description of hashmap.
- Have a syntax pseudo block for hashmap.
- Have an example code block for each hashmap.

### Entry Type (select all that apply)

New Topic Entry

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [x] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by dakshdeepHERE**
### Type of Change

- Adding a new entry
- Updating the documentation

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

Fixes #1477

### CODE REVIEW

File: content/kotlin/concepts/hashmap/hashmap.md
Comment (SSwiniarski) [inline content/kotlin/concepts/hashmap/hashmap.md:3]:
```suggestion
Description: 'A HashMap is used to store items as a key-value pairs. The keys and values can be of either same or different types.'
```

File: content/kotlin/concepts/hashmap/hashmap.md
Comment (SSwiniarski) [inline content/kotlin/concepts/hashmap/hashmap.md:26]:
```suggestion
The types can also be specified when creating an instance of the `HashMap` class like this:
```

File: content/kotlin/concepts/hashmap/hashmap.md
Comment (SSwiniarski) [inline content/kotlin/concepts/hashmap/hashmap.md:32]:
```suggestion
The `mutableMapOf()` function is used to create a mutable HashMap:
```

File: content/kotlin/concepts/hashmap/hashmap.md
Comment (SSwiniarski) [inline content/kotlin/concepts/hashmap/hashmap.md:38]:
```suggestion
The `mapOf()` function is used to create an immutable `HashMap`:
```

File: content/kotlin/concepts/hashmap/hashmap.md
Comment (SSwiniarski) [inline content/kotlin/concepts/hashmap/hashmap.md:46]:
```suggestion
To access the elements of a `HashMap`, the `get()` function is used to retrieve the value for a given key. For example:
```

File: content/kotlin/concepts/hashmap/hashmap.md
Comment (SSwiniarski) [inline content/kotlin/concepts/hashmap/hashmap.md:66]:
```suggestion
The `containsKey()` function is used to check if a HashMap contains a given key:
```

File: content/kotlin/concepts/hashmap/hashmap.md
Comment (SSwiniarski) [inline content/kotlin/concepts/hashmap/hashmap.md:74]:
```suggestion
The `forEach()` function is used to iterate over the key-value pairs in the HashMap:
```

File: content/kotlin/concepts/hashmap/hashmap.md
Comment (SSwiniarski) [inline content/kotlin/concepts/hashmap/hashmap.md:84]:
```suggestion
To add a key-value pair to a `HashMap` in Kotlin, the `put()` function is used. For example:
```

File: content/kotlin/concepts/hashmap/hashmap.md
Comment (SSwiniarski) [inline content/kotlin/concepts/hashmap/hashmap.md:94]:
```suggestion
The index operator `[]` can also be used to add a key-value pair to the `HashMap`. For example:
```

File: content/kotlin/concepts/hashmap/hashmap.md
Comment (SSwiniarski) [inline content/kotlin/concepts/hashmap/hashmap.md:104]:
```suggestion
The `putAll()` function can be used to add all the key-value pairs from another map to the `HashMap`:
```

File: content/kotlin/concepts/hashmap/hashmap.md
Comment (SSwiniarski) [inline content/kotlin/concepts/hashmap/hashmap.md:122]:
```suggestion
To remove a key-value pair from a `HashMap` in Kotlin, the `remove()` function is used. For example:
```

File: content/kotlin/concepts/hashmap/hashmap.md
Comment (SSwiniarski) [inline content/kotlin/concepts/hashmap/hashmap.md:134]:
```suggestion
The `remove()` function can be used to remove a key-value pair from the `HashMap` only if the value matches a certain condition. For example:
```

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@dakshdeepHERE, I did a first review and left a few comments. Most of which were to remove direct address to the reader. In Docs we prefer not to use constructions such as "you can do this."

**Comment by dakshdeepHERE** [comment]
@SSwiniarski I'm really sorry I always forget to not write "you" because I have never wrote a doc before I will try my best in my next pr !

**Comment by dakshdeepHERE** [comment]
@SSwiniarski you can check I did all the commits also can we do anything about my old pr? Which are done with both the reviews and waiting for the author?

**Review by SSwiniarski — APPROVED** [review]
@dakshdeepHERE Looks good for a second review. As for your other PRs, a maintainer will get to them for further reviews, just have some patience.

File: content/kotlin/concepts/hashmap/hashmap.md
Comment (Dusch4593) [reply content/kotlin/concepts/hashmap/hashmap.md:21]:
```suggestion
val map = HashMap<KeyType, ValueType>()
val map: HashMap<KeyType, ValueType> = HashMap()
```

I would at least consolidate this line with the pseudocode on line # 29.

File: content/kotlin/concepts/hashmap/hashmap.md
Comment (Dusch4593) [inline content/kotlin/concepts/hashmap/hashmap.md:30]:
```suggestion
The `KeyType` and `ValueType` are the data types respectively assigned to the keys and values of the `map`.
```

File: content/kotlin/concepts/hashmap/hashmap.md
Comment (Dusch4593) [inline content/kotlin/concepts/hashmap/hashmap.md:42]: I don't have much experience with Kotlin, but I had a few questions about these lines.

- Isn't the `mutableMapOf()` more closely related to `MutableMap` (which is the interface for `HashMap`)?
- Doesn't the `mapOf()` have more to do with the `Map` interface?

These questions came to mind after doing a little research through [the official Kotlin docs](https://kotlinlang.org/docs/home.html).

**Review by Dusch4593 — CHANGES_REQUESTED** [review]
@dakshdeepHERE I left you some comments/suggestions but I may have more after this feedback is implemented. Additionally, I think that all the content from line # 49 through the rest of this file should be separated out into separate term entries under this topic for the following methods/functions:

- `put()`
- `.putAll()`
- `.containsKey()`
- `.forEach()`
- `.remove()`

This would be 5 additional entries. You could either add them here in this PR, or split them up however you like and create another PR or two. 😄

**Comment by dakshdeepHERE** [comment]
Alright I'll delete those lines and open separate pr for the term folder.

**Comment by Dusch4593** [comment]
> Alright I'll delete those lines and open separate pr for the term folder.

@dakshdeepHERE Let us know when you've deleted those lines. We should merge this PR first before the other PR gets created.

File: content/kotlin/concepts/hashmap/hashmap.md
Comment (dakshdeepHERE) [reply content/kotlin/concepts/hashmap/hashmap.md:37]: Damm I'm impressed that you asked this question just by reading the docs from the site. I usually learn stuff from Coursera. But I'm trying to read the docs too. Specially the jetbrains doc for kotlin is amazing. 
- Yes, you are right. The `mutableMapOf()` function in Kotlin creates an instance of a `MutableMap`, which is an interface that is implemented by the HashMap class. This function is a shorthand for creating a new `HashMap` and populating it with key-value pairs.
- I think it is not directly related to the Map interface in, although it can be used to create a map object that implements the Map interface.

**Comment by dakshdeepHERE** [comment]
@Dusch4593 please check and btw should I remove the get() too from this file?

**Comment by Dusch4593** [comment]
> @Dusch4593 please check and btw should I remove the get() too from this file?

Yeah, let's remove the content pertaining to `get()` as well and move to its own term entry. After this is done, I'll take another look. 😄

**Comment by dakshdeepHERE** [comment]
@Dusch4593 alright wait I will remove get() too.

**Comment by Dusch4593** [comment]
> @Dusch4593 done 👍🏻

Thank you very much for doing that, @dakshdeepHERE! 🚀 

I just took a look and it seems that you resolved some comments/suggestions made my myself and @SSwiniarski without implementing any changes. Some of this suggestions should be implemented because some of the lines include "you", and Docs should maintain a more neutral, instructive tone; never addressing any reader(s) directly.

Let me know when you've gone through these and I'll take another look. 😄

**Comment by dakshdeepHERE** [comment]
@Dusch4593 I think this happened because I forced pushed my commits on this branch because when I try to push normally I get this error 
image

**Comment by dakshdeepHERE** [comment]
I remember I changed all the commits that u requested

**Review by Dusch4593 — APPROVED** [review]
@dakshdeepHERE LGTM for merge! 😄 

I added an "## Example" section in 3d9a6dee2d92405c09b1b1e497e4489a32376612 and c6cd2fc1bc435b23fb6653ba56bb1f01b37b1db5, with a \```kotlin code block, that creates a small `HashMap` of country-capital pairs and prints them out.

**Comment by Dusch4593** [comment]
Thank you for another wonderful contribution to Codecademy Docs, @dakshdeepHERE! 🚀 🎊 🎉

The new entry should now be live: https://www.codecademy.com/resources/docs/kotlin/hashmap

Screen Shot 2023-01-22 at 1 27 27 PM

Looking forward to your next Docs contribution! 😄

**Comment by dakshdeepHERE** [comment]
@yangc95 can you list all the changes that you want the in folder hashmap?

**Comment by yangc95** [comment]
Thanks @dakshdeepHERE! Other than renaming the folder a file here, you can refer to this comment for the term that's under the hashmap folder: https://github.com/Codecademy/docs/pull/1517#issuecomment-1402300813

It's great to see more terms being added here.

**Comment by dakshdeepHERE** [comment]
@yangc95 ok then I can just open the `.get()` pr again and change all the mentions to `.get()`. Right?

**Comment by lucwhite** [comment]
Hello @dakshdeepHERE ,

I hope this message finds you well! My name is Lucas, and I work on the Curriculum team at Codecademy. Our team is considering ways to encourage more individuals to contribute to Codecademy Docs and improve the contribution experience. I was wondering if you would be willing to participate in an interview to provide information on your background and experience with Docs contribution?

The interview would take about an hour and be conducted over Zoom. Should you agree to participate, we will find a time window using Calendly that works for you within the next three weeks. Let us know if you are interested in participating in an interview or have any questions by emailing me at the address below!

Best,
Lucas White
[lucas@codecademy.com](mailto:lucas@codecademy.com)

**Comment by dakshdeepHERE** [comment]
yes sure @lucwhite I always love contributing in this repo as it helps me a lot with my learning also all the awsm mentors here helped me with a lot of problems with git and in `java` and `kotlin`. I will mail you right away I'm really sorry for the delayed reply as this week the college was very stressful tons of tests and projects to submit.

**Pull request merged**
